packages/forms-web-app/src/dynamic-forms/dynamic-components/address-add-more/question.js

```diff
@@ -25,16 +28,24 @@ class AddressAddMoreQuestion extends AddMoreQuestion {
 	/**
 	 * adds a uuid and an address object for save data using req body fields
 	 * @param {ExpressRequest} req
+	 * @param {JourneyResponse} journeyResponse - current journey response, modified with the new answers
 	 * @returns
 	 */
-	async getDataToSave(req) {
+	async getDataToSave(req, journeyResponse) {
 		const address = new Address({
 			addressLine1: req.body[this.fieldName + '_addressLine1'],
 			addressLine2: req.body[this.fieldName + '_addressLine2'],
 			townCity: req.body[this.fieldName + '_townCity'],
 			postcode: req.body[this.fieldName + '_postcode']
 		});
-		return { addMoreId: uuid.v4(), value: address };
+
+		await apiClient.postSubmissionNeighbourAddress(
+			journeyResponse.referenceId,
+			journeyResponse.answers.id,
+			address
+		);
```

Comment:
Is this the right place? The existing function just returns some data, it doesn't save it?

Reply:
moved